tests/assert_type/db/test_transaction.py

```diff
@@ -0,0 +1,33 @@
+from collections.abc import Callable
+
+from django.db.transaction import atomic, non_atomic_requests
+from django.http import HttpRequest, HttpResponse
+from typing_extensions import assert_type
+
+
+@atomic
+def func1(x: int, /) -> list[int]: ...
+
+
+assert_type(func1, Callable[[int], list[int]])
+
+
+@atomic(using="bla", savepoint=False)
+def func2(x: int, /) -> list[int]: ...
+
+
+assert_type(func2, Callable[[int], list[int]])
```

Comment:
this does not differentiate `(x: int)` and `(x: int, /)`. A Protocol, maybe?

Reply:
`assert_type` doesn't work without forced positional, as it doesn't seem to be possible to do something like:

```python
assert_type(func2, Callable[[Arg(int, "x")], list[int]]
```

Reply:
Ah, alright, lets just avoid `assert_type` for the Callables and it should be fine I think. Could update to something like:

```python
x: Callable[[int], list[int]] = func2
```

Ref: https://github.com/python/mypy/issues/13935